### ISSUE TITLE
Code completion does not list macro after perform go-to-def on macro.

### ISSUE THREAD

**Issue by lh123** [clangd/clangd#478]
**System information**
Output of `clangd --version`: `clangd version 12.0.0 (https://github.com/llvm/llvm-project.git 8dc820393219c7ee440b4ec86c9a201301943276)`
Editor/LSP plugin: VSCode
Operating system: Win10
Gcc: `gcc version 10.1.0 (Rev3, Built by MSYS2 project)`

1. create `main.cpp`
```c++
#include <windows.h>

int main() {
    WS_CHILD;
}
```
2. perform go-to-def on `WS_CHILD`.
3. trigger code complete for `WS_CHIL`.
![image](https://user-images.githubusercontent.com/10697249/88757962-e3c8a380-d199-11ea-98d6-ea014f574efe.png)

full log: 
[log.txt](https://github.com/clangd/clangd/files/4992657/log.txt)

**Comment by kadircet** [comment]
How is this any different than https://github.com/clangd/clangd/issues/461 ?

Unfortunately i cannot repro as I don't have access to windows.h, could you try to provide a minimal repro ?

**Comment by lh123** [comment]
Yes, this is very similar to #461, but #461 has been fixed in https://github.com/llvm/llvm-project/commit/a69f9a8584f2a090b5fe6235a112f9b68c324863.

I guess the difference is that the header file `winuser.h` containing WS_CHILD is not a self-contained file which cause clang report `Too many errors emitted, stopping now`.

**Comment by kadircet** [comment]
Ah right that makes sense. I am not sure what we can do about this though, it boils down to #45

**Comment by lh123** [comment]
It doesn't repro with `non-self-contained` file.
1. compile_flags.txt
```
-ferror-limit=5
-xc++
```
2. main.cpp
```c++
#define TEST
typedef int mint;
#include "test.h"

int main() {
    MY_MACRO;
}

```
3. test.h
```c++
void test(mint);
void test1(mint);
void test2(mint);
void test3(mint);
void test4(mint);
void test5(mint);

#ifdef TEST
#define MY_MACRO
#endif
```

`windows.h` is too complicated, I can’t get the minimal repro from it

**Comment by kadircet** [comment]
it actually does, e.g.:

a.h:
```
#ifndef X
#define YESX
#else
#define NOX
#endif
```

a.cc:
```
#define X
#include "a.h"

Y^
```

triggering code completion at carrot would yield `YESX` if you haven't opened a.h yet, but once you open it code completion won't yield any info (but now you can complete for `NOX`).
